### PR TITLE
Fixes and better documentation

### PR DESCRIPTION
**Pull request by SoffiMeower**
Adds safer/less error prone refactoring, more verbose logging, and fixes the list reader.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,37 @@
 # Auth Info Dumper
 
-This is a PS4 payload to dump [auth info](https://www.psdevwiki.com/ps4/Auth_Info) from a SELF file. Right now it is hard-coded file name to dump and it is only work on 9.0.
+This is a PS4 payload to dump [auth info](https://www.psdevwiki.com/ps4/Auth_Info) from a ELF, SELF, or eboot.bin file. This payload only works on firmware 9.00!
+
+## How to use
+
+1. Download .bin from [Github Actions](test) or build it yourself!
```
Comment (ultimaweapon) [inline README.md]: Better to remove the Actions link until it is actually available.

File: source/main.c [hunk 2]
```diff
@@ -36,108 +36,114 @@ static int get_auth(struct thread *td, struct get_auth_args *args) {
...
+        if (kexec(get_auth, &req) < 0) {
+            printf_notification("Failed to dump auth info for\n%s", next);
+            free(buf);
+            return 0;
+        }
+
+        next = p + 1;
+        p = strchr(next, '\n');
+        if (p) {
+            if (p > next && p[-1] == '\r') {
```
Comment (ultimaweapon) [inline source/main.c]: Why check `p > next`?

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (SoffiMeower) [reply README.md]: DID I NOT REMOVE THAT-

File: source/main.c (hunk 2 above)
Comment (SoffiMeower) [reply source/main.c]: Basically just a safety double check to ensure we don't read memory before the string starts by accident.

Comment (ultimaweapon) [reply source/main.c]: Where did you get this code from?

Comment (SoffiMeower) [reply source/main.c]: Eh, mainly just knowing how unsafe C code can be, I was just adding anything I could to make sure it didn't break lol.

Comment (SoffiMeower) [reply source/main.c]: Yeah, I'm removing that check... That was kind of idiotic on my part XD... A double check made me kinda realize "Oh, wait, that won't happen."

Comment (ultimaweapon) [reply source/main.c]: Who are written this code? You or someone else?

Comment (SoffiMeower) [reply source/main.c]: > Who are written this code? You or someone else?

Kinda both, I mainly was looking online for ways to try to make sure our code was as safe as possible, and implemented as I went along.

File: README.md [hunk 3]
```diff
@@ -1,14 +1,37 @@
...
+## How to use
+
+1. Download .bin from [Releases](https://github.com/obhq/auth-dumper/releases/latest) or build it yourself!
+2. Setup your prefered payload loader, for Payload Guest, put the payload on the root of your usb under a folder named `payloads` or on your PS4 under `/data/payloads`
+3. On the root of your USB, make a file named `dump-list.txt`
+4. This payload can dump multiple auth-infos on one execution, so open `dump-list.txt` and format each line like this: `/readauthinfofrom/folder/file.self:/exportauthinfoto/file.bin` An example is below:
+```
+/system/sys/SceSysCore.elf:/data/dumpedSysCore.bin
+/system/sys/SceVdecProxy.elf:/data/dumpedVdecProxy.bin
+```
```
Comment (ultimaweapon) [inline README.md:14]: This format broke the rendered list.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (SoffiMeower) [reply README.md:14]: Uh, no? That two line list was meant to be in a code block... Which it is.

Comment (ultimaweapon) [reply README.md:14]: Screenshot 2023-12-08 at 23-21-54 auth-dumper_README md at 9df164978d22f5dd22be63dceb2f82e247b3c52c · obhq_auth-dumper

Comment (SoffiMeower) [reply README.md:14]: There, that should fix it.

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**